cmd/shortener/main.go

```diff
@@ -1,18 +1,58 @@
 package main
 
 import (
-	"github.com/julienschmidt/httprouter"
+	"context"
+	"github.com/DelusionTea/praktikum-go/cmd/conf"
 	"github.com/DelusionTea/praktikum-go/internal/app/handlers"
+	"github.com/DelusionTea/praktikum-go/internal/memory"
+	"github.com/gin-gonic/gin"
 	"log"
 	"net/http"
+	"os"
+	"os/signal"
 )
 
+func setupRouter(repo memory.MemoryInterface, baseURL string) *gin.Engine {
+	router := gin.Default()
+
+	handler := handlers.New(repo, baseURL)
+
+	router.GET("/:id", handler.HandlerGetURLByID)
+	router.POST("/", handler.HandlerCreateShortURL)
+	router.POST("/api/shorten", handler.HandlerShortenURL)
+
+	router.HandleMethodNotAllowed = true
+
+	return router
+}
+
 func main() {
-	//http.HandleFunc("/", myrequest)
-	router := httprouter.New()
-	router.POST("/", handlers.HandlerCreateShortURL)
-	router.GET("/:id", handlers.HandlerGetURLByID)
-	//Сервер должен быть доступен по адресу: http://localhost:8080.
-	//http.ListenAndServe(":8080", nil)
-	log.Fatal(http.ListenAndServe("localhost:8080", router))
+	cfg := conf.GetConfig()
+
+	if string(cfg.BaseURL[len(cfg.BaseURL)-1]) != "/" {
```

Comment:
Не понимаю зачем к строке приводить? Можно с отдельным символом сработать. Но тут у тебя бага может быть, если BaseUrl это пустая строка: будешь к индексу -1 обращаться и панику вызовешь.
```suggestion
	if cfg.BaseURL[len(cfg.BaseURL)-1] != '/' {
```